locust_plugins/listeners.py

```diff
@@ -130,22 +135,36 @@ def on_start(self, environment: locust.env.Environment):
             sys.exit(1)
         self.set_gitrepo()
 
-        if is_worker() or is_master():
+        if (
+            self.env.parsed_options.worker
+            or self.env.parsed_options.master
+            or isinstance(environment.runner, LocalRunner)
+        ):
             # swarm generates the run id for its master and workers
             if getattr(environment.parsed_options, "run_id", False):
                 self._run_id = parser.parse(environment.parsed_options.run_id)
+            elif hasattr(self, "_run_id"):
+                pass
             else:
                 logging.info(
                     "You are running distributed, but without swarm. run_id:s in Timescale will not match exactly between load gens"
                 )
                 self._run_id = datetime.now(timezone.utc)
+                logging.info(f"Run id is {self._run_id}")
```

Comment:
Shortly after this, the url will be logged anyway. Maybe make this debug level?

Reply:
updated.